swig/include/python/gdal_python.i

```diff
@@ -549,8 +553,72 @@ void wrapper_VSIGetMemFileBuffer(const char *utf8_path, GByte **out, vsi_l_offse
                   resample_alg=gdalconst.GRIORA_NearestNeighbour,
                   callback=None,
                   callback_data=None):
-      """ Reading a chunk of a GDAL band into a numpy array. The optional (buf_xsize,buf_ysize,buf_type)
-      parameters should generally not be specified if buf_obj is specified. The array is returned"""
+      """
+      Read a window of this raster band into a NumPy array.
+
+      Parameters
+      ----------
+      xoff : int, default=0
```

Comment:
xoff/yoff could be float too. Applies to other similar calls

Reply:
I updated the `ReadAsArray` methods to use `float` here and show an example of its use. On the `WriteArray` side the behavior I'm seeing is the same as rounding a float offset to an integer, so I've left it as `int`.